src/main/kotlin/parts/wisdom/arcraider/visualization/Grid.kt

```diff
@@ -0,0 +1,28 @@
+package parts.wisdom.arcraider.visualization
+
+import parts.wisdom.arcraider.SerializedGrid
+
+data class Coord( val x : Int, val y : Int)
+
+open class Grid(
+    val widthSquares: Int,
+    val heightSquares: Int,
+    val coords: List<Coord>) {
+
+    constructor(serializedGrid : SerializedGrid) :
+            this(serializedGrid[0].size, serializedGrid.size, generateCoords(serializedGrid))
+}
+
+
+fun generateCoords(serializedGrid: SerializedGrid): MutableList<Coord> {
```

Comment:
I'd lean toward returning `List<Coord>`, but I can see that both ways. The function *does* construct a `MutableList`, so one could argue it's made more versatile at zero cost by confessing to that in the return value. I just have a strong enough bias toward immutable data structures that I stumble over the mutable return value and ask myself "huh -- why mutable?".